src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -274,11 +274,12 @@ case class AppDefinition(
     )
   }
 
-  private val portIndicies: Range = container.flatMap(_.hostPorts).getOrElse(portNumbers).indices
+  private val portIndices: Range = container.flatMap(_.hostPorts.filter(_.nonEmpty)).getOrElse(portNumbers).indices
```

Comment:
Proto has no notion of an "optional repeated field" its also very strange to have an Option[Seq[]] of anything. Option is by definition a Seq of 0 or 1 element. I tried to keep this and the other regression change as minimal as I could - keeping the Option[Seq[]] in place - but fixing the serialization/deserialization (there were a lot of regressions). I have a branch at home where I started to remove the wrapping Option[] but it didn't help at all.